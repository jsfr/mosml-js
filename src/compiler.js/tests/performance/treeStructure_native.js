"use strict"

var ctimeBegin$1 = function($_var1) {
    return console.time($_var1);
};
var ctimeEnd$2 = function($_var5) {
    return console.timeEnd($_var5);
};
var repeat$3 = function($_var9) {
    return function($_var10) {
        return function($_var11) {
            var $_var13 = function($_var14) {
                return (function() {
                    switch ($_var14) {
                        case 0:
                            return $_var10($_var11)
                        default:
                            return ($_var10($_var11), $_var13($_var14 - 1))
                    }
                }());
            };
            return $_var13($_var9 - 1);
        };
    };
};
var timerep$4 = function($_var35) {
    return function($_var36) {
        return function($_var37) {
            return repeat$3($_var35)(function($_var41) {
                return (ctimeBegin$1($_var37), ($_var36($_var41), ctimeEnd$2($_var37)));
            });
        };
    };
};

var Tree = function(val, left, right){
    var instance = {};
    instance.val = val;
    instance.leftTree = left;
    instance.rightTree = right;
    instance.tag = "Tree";
    return instance;
}
var Leaf = function(){
    var instance = {};
    instance.tag = "Leaf";
    return instance;
}

var intList = function(n) {
    var list = [];
    while (n >= 0) {
        list.push(n)
        --n;
    }
    return list;
}

var rev = function(list) {
    var revlist = [];
    var n = list.length;
    while (n-- > 0) {
        revlist.push(list[n]);
    }
    return revlist;
}

var genTreeMake = function(n) {
    var nums = intList(19);
    return genTree(nums);
}

var genTree = function(nums) {
    switch(nums.length) {
        case 0:
            return Leaf();
        default:
            var head = nums.shift();
            var revlist = rev(nums)
            return Tree(head, genTree(nums), genTree(revlist));
    }
}

var traverseTree = function(tree, n) {
    switch (tree.tag) {
        case "Tree":
            return (n == tree.val ? 1 : 0) + traverseTree(tree.leftTree, n) + traverseTree(tree.rightTree, n);
        default:
            return 0;
    }
}

var tree = genTree(intList(19));
var $_var154 = timerep$4(100)(genTreeMake)("genTree_native")(19);
var $_var171 = timerep$4(100)(traverseTree)("traverseTree_native")(tree, 0);