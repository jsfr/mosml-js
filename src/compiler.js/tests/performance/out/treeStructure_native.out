==============================JS BEGIN==============================
genTree_native: 702ms
genTree_native: 717ms
genTree_native: 964ms
genTree_native: 797ms
genTree_native: 841ms
genTree_native: 706ms
genTree_native: 914ms
genTree_native: 679ms
genTree_native: 915ms
genTree_native: 854ms
genTree_native: 663ms
genTree_native: 785ms
genTree_native: 1008ms
genTree_native: 809ms
genTree_native: 646ms
genTree_native: 793ms
genTree_native: 904ms
genTree_native: 676ms
genTree_native: 922ms
genTree_native: 818ms
genTree_native: 650ms
genTree_native: 793ms
genTree_native: 962ms
genTree_native: 811ms
genTree_native: 662ms
genTree_native: 826ms
genTree_native: 913ms
genTree_native: 846ms
genTree_native: 650ms
genTree_native: 795ms
genTree_native: 919ms
genTree_native: 802ms
genTree_native: 661ms
genTree_native: 795ms
genTree_native: 925ms
genTree_native: 822ms
genTree_native: 642ms
genTree_native: 817ms
genTree_native: 917ms
genTree_native: 822ms
genTree_native: 655ms
genTree_native: 809ms
genTree_native: 971ms
genTree_native: 810ms
genTree_native: 644ms
genTree_native: 790ms
genTree_native: 912ms
genTree_native: 874ms
genTree_native: 651ms
genTree_native: 796ms
genTree_native: 939ms
genTree_native: 818ms
genTree_native: 700ms
genTree_native: 781ms
genTree_native: 963ms
genTree_native: 653ms
genTree_native: 904ms
genTree_native: 843ms
genTree_native: 635ms
genTree_native: 809ms
genTree_native: 915ms
genTree_native: 805ms
genTree_native: 660ms
genTree_native: 791ms
genTree_native: 924ms
genTree_native: 806ms
genTree_native: 638ms
genTree_native: 814ms
genTree_native: 906ms
genTree_native: 823ms
genTree_native: 634ms
genTree_native: 789ms
genTree_native: 944ms
genTree_native: 811ms
genTree_native: 678ms
genTree_native: 818ms
genTree_native: 913ms
genTree_native: 912ms
genTree_native: 649ms
genTree_native: 909ms
genTree_native: 914ms
genTree_native: 821ms
genTree_native: 681ms
genTree_native: 806ms
genTree_native: 956ms
genTree_native: 657ms
genTree_native: 936ms
genTree_native: 853ms
genTree_native: 639ms
genTree_native: 803ms
genTree_native: 923ms
genTree_native: 813ms
genTree_native: 654ms
genTree_native: 777ms
genTree_native: 932ms
genTree_native: 815ms
genTree_native: 639ms
genTree_native: 809ms
genTree_native: 904ms
genTree_native: 823ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 42ms
traverseTree_native: 41ms
traverseTree_native: 42ms
traverseTree_native: 42ms
traverseTree_native: 42ms
traverseTree_native: 44ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 39ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 42ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 39ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 39ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 79ms
traverseTree_native: 60ms
traverseTree_native: 44ms
traverseTree_native: 44ms
traverseTree_native: 49ms
traverseTree_native: 41ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
traverseTree_native: 40ms
==============================JS END================================
